docs/styling-with-css/index.md

```diff
@@ -0,0 +1,322 @@
+project are visible to all other modules[^2]. However, we can make use of
+JavaScript's `import` syntax by using the [mel.raw extension
+node](https://melange.re/v2.1.0/communicate-with-javascript/#generate-raw-javascript),
+which allows us to embed raw JavaScript in our OCaml code. Add the following
+line to the top of `Order.re`:
+
+```reason
+[%%mel.raw {|import "./order-item.css"|}];
+```
+
+The `{||}` string doesn't require you to escape characters like double-quote and
+newline. They are similar to the `{js||js}` strings we first saw in the [Celsius
+Converter](/celsius-converter-exception/#solutions) chapter, with the
+difference that they won't handle Unicode correctly.
+
+Unfortunately, in the terminal where we're running `make serve`, we see this
+webpack compilation error:
+
+```
+ERROR in ./_build/default/src/order-confirmation/output/src/order-confirmation/Order.js 7:0-8:1
+Module not found: Error: Can't resolve './order-item.css' in '~/melange-for-react-devs/_build/default/src/order-confirmation/output/src/order-confirmation'
+```
+
+## Tell Dune to copy CSS files
+
+The problem is that webpack is serving the app from the build directory at
+`_build/default/src/order-confirmation/output/src/order-confirmation`, and the
+`order-item.css` file isn't in the build directory.
+
+To solve this, we can add the [runtime_deps
+field](https://melange.re/v2.1.0/build-system/#handling-assets) to our
+`melange.emit` stanza in `src/order-confirmation/dune`:
+
+```clj{7}
+(melange.emit
+ (target output)
+ (libraries reason-react)
+ (preprocess
+  (pps melange.ppx reason-react-ppx))
+ (module_systems es6)
+ (runtime_deps order-item.css))
+```
+
+We also want to add styles for the `Order` component, so add a new file
+`src/order-confirmation/order.css` with these styles:
+
+<!-- This file gets renamed to order.module.css later -->
+<<< @/../src/styling-with-css/order.module.css
+
+To ensure that `order.css` is also copied to the build directory, we can add
+`order.css` to the value of `runtime_deps`:
+
+```clj
+(runtime_deps item.css order.css)
+```
+
+If you have many `.css` files, you can just tell `runtime_deps` to copy all
+`.css` files over:
+
+```clj
+(runtime_deps (glob_files *.css))
+```
+
+## Add classes to JSX
+
+Now we can add the appropriate classes to `OrderItem.make`'s JSX:
+
+<<< Order.re#order-item{4-6}
+
+As well as `Order.make`'s JSX:
+
+<<< Order.re#make{6,13}
+
+Finally, add a `mel.raw` extension node at the top of `Order.re`:
+
+```reason
+[%%mel.raw {|import "./order.css"|}];
+```
+
+## Problems with `mel.raw`
+
+This solution works well for our current build configuration, but falls apart if
+we change the `module_systems` field of the `melange.emit` stanza from `es6` to
+`commonjs`. This results in a subtle runtime error caused by the fact
+that CommonJS uses `require` instead of `import` to import modules.
+
+Another issue is that you can't easily use `mel.raw` extension nodes in
```

Comment:
It turns out that Melange allows you to put `%%mel.raw` inside a submodule, but refmt will always reformat it to `%mel.raw` in this case, which causes a compilation error in webpack. Not sure if I should include more detail about that here.